spec/money_spec.rb

```diff
@@ -149,58 +149,89 @@
   end
 
   describe ".from_amount" do
-    it "accepts numeric values" do
-      expect(Money.from_amount(1, "USD")).to eq Money.new(1_00, "USD")
-      expect(Money.from_amount(1.0, "USD")).to eq Money.new(1_00, "USD")
-      expect(Money.from_amount("1".to_d, "USD")).to eq Money.new(1_00, "USD")
-    end
-
     it "raises ArgumentError with unsupported argument" do
       expect { Money.from_amount("1") }.to raise_error(ArgumentError)
       expect { Money.from_amount(Object.new) }.to raise_error(ArgumentError)
     end
 
-    it "converts given amount to subunits according to currency" do
-      expect(Money.from_amount(1, "USD")).to eq Money.new(1_00, "USD")
-      expect(Money.from_amount(1, "TND")).to eq Money.new(1_000, "TND")
-      expect(Money.from_amount(1, "JPY")).to eq Money.new(1, "JPY")
-    end
+    context 'with default currency', :default_currency do
```

Comment:
Looks good, but can you also include a case when the default currency is `nil` and someone tries to call `from_amount` without a currency

Reply:
Test for that specific case was added. 👌